### PR TITLE
Fix 2756

### LINKED ISSUE #2756: String.toLowerCase() produces error on strings with Umlaut.
When using the .toLowerCase() (or the .toLocalLowerCase()) function on strings containing Umlauts, the design document view will not produce any results.

## Description

See above.

## Steps to Reproduce

1. Create a database and add any document to it.
1. Creata a design document with a view and use the following code
```
function (doc) {
  emit('Ä'.toLowerCase(),1);
}
```

## Expected Behaviour

A view will be created with at least one entry (as many as there were documents) with `key: 'ä'` and `value: 1`.

## Your Environment

* CouchDB version used: 3.0.0 dockerized and pulled from dockerhub (couchdb:latest)
* Browser name and version: Firefox 74.0
* Operating system and version: docker on Xubuntu 19.10

## Additional Context

[TIP]:  # ( Add any other context about the problem here. )

### PR DESCRIPTION
**Pull request by florianb**
# TBD

## Overview

Fixes #2756.

## Checklist

- [ ] Code is written and works correctly
- [ ] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/couch/test/eunit/couchdb_regression_2756_tests.erl [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+% the License.
+
+-module(couchdb_regression_2756_tests).
+
+-include_lib("couch/include/couch_eunit.hrl").
+-include_lib("couch/include/couch_db.hrl").
+
+setup() ->
+    DbName = ?tempdb(),
+    {ok, Db} = couch_db:create(DbName, [?ADMIN_CTX]),
```
Comment (florianb) [inline src/couch/test/eunit/couchdb_regression_2756_tests.erl:20]:
```
Compiled test/eunit/couchdb_regression_2756_tests.erl
    Running test function(s):
      couchdb_regression_2756_tests:should_lower_case_single_umlaut_test/0
======================== EUnit ========================
couchdb_regression_2756_tests: should_lower_case_single_umlaut_test (module 'couchdb_regression_2756_tests')...*failed*
in function couch_epi_functions_gen_couch_db:providers/2
  called as providers(validate_dbname,2)
in call from couch_epi_functions_gen:decide/5 (src/couch_epi_functions_gen.erl, line 62)
in call from couch_db_plugin:maybe_handle/3 (src/couch_db_plugin.erl, line 89)
in call from couch_server:validate_open_or_create/2 (src/couch_server.erl, line 698)
in call from couch_server:create/2 (src/couch_server.erl, line 128)
in call from couchdb_regression_2756_tests:setup/0 (test/eunit/couchdb_regression_2756_tests.erl, line 20)
in call from couchdb_regression_2756_tests:should_lower_case_single_umlaut_test/0 (test/eunit/couchdb_regression_2756_tests.erl, line 33)
in call from couchdb_regression_2756_tests:should_lower_case_single_umlaut_test/0
**error:undef
  output:<<"">>
```

File: src/couch/priv/couch_js/60/main.cpp [hunk 2]
```diff
@@ -412,8 +412,10 @@ main(int argc, const char* argv[])
 
     couch_args* args = couch_parse_args(argc, argv);
 
-    JS_Init();
-    cx = JS_NewContext(args->stack_size, 8L * 1024L);
+    if(!JS_Init())
+        return 1;
```
Comment (florianb) [inline src/couch/priv/couch_js/60/main.cpp:416]: I added this to be sure the initialization didn't fail silently.

File: src/couch/priv/couch_js/60/main.cpp [hunk 3]
```diff
@@ -432,6 +434,9 @@ main(int argc, const char* argv[])
     JS::CompartmentOptions options;
     JS::RootedObject global(cx, JS_NewGlobalObject(cx, &global_class, nullptr,
                                                    JS::FireOnNewGlobalHook, options));
+
+    JS_SetDefaultLocale(JS_GetRuntime(cx), "en");
```
Comment (florianb) [inline src/couch/priv/couch_js/60/main.cpp:438]: That's the experimental fix, couldn't proove it yet, and the use of `JS_GetRuntime` might not be the right way. Probably a dedicated Runtime-object must be created?

**Comment by wohali** [comment]
@florianb You may want to see #2786 which has all of @davisp 's work in it on this, which includes both your and my feedback.

It's quite a bit of work!

**Comment by florianb** [comment]
This is obsolete due to #2786

**Comment by florianb** [comment]
Thanks @wohali - #2786 solves the issue. If i might lend a helping hand, let me know.

**Pull request closed without merging**